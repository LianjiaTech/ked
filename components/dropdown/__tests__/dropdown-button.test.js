import React from 'react';
import { mount } from 'enzyme';
import Dropdown from '../dropdown';
import Menu from '../../menu';
import mountTest from '../../../tests/shared/mountTest';

describe('DropdownButton', () => {
  mountTest(() => (
    <Dropdown menu={<Menu />}>
      <span />
    </Dropdown>
  ));
  mountTest(Dropdown.Button);

  it('pass appropriate props to Dropdown', () => {
    const props = {
      align: {
        offset: [10, 20],
      },
      overlay: (
        <Menu>
          <Menu.Item>foo</Menu.Item>
        </Menu>
      ),
      disabled: false,
      trigger: ['hover'],
      visible: true,
      onVisibleChange: () => {},
    };

    const wrapper = mount(<Dropdown.Button {...props} />);
    const dropdownProps = wrapper.find(Dropdown).props();

    Object.keys(props).forEach(key => {
      expect(dropdownProps[key]).toBe(props[key]); // eslint-disable-line
    });
  });

  it("don't pass visible to Dropdown if it's not exits", () => {
    const menu = (
      <Menu>
        <Menu.Item>foo</Menu.Item>
      </Menu>
    );
    const wrapper = mount(<Dropdown.Button overlay={menu} />);
    const dropdownProps = wrapper.find(Dropdown).props();

    expect('visible' in dropdownProps).toBe(false);
  });

  it('should support href like Button', () => {
    const menu = (
      <Menu>
        <Menu.Item>foo</Menu.Item>
      </Menu>
    );
    const wrapper = mount(<Dropdown.Button overlay={menu} href="https://ant.design" />);
    expect(wrapper.render()).toMatchSnapshot();
  });
});
