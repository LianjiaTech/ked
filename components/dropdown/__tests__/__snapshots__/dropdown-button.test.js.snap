// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropdownButton should support href like Button 1`] = `
<div
  class="ant-btn-group ant-dropdown-button"
>
  <a
    class="ant-btn ant-btn-default"
    href="https://ant.design"
  />
  <button
    class="ant-btn ant-dropdown-trigger ant-btn-default"
    type="button"
  >
    <i
      aria-label="icon: ellipsis"
      class="anticon anticon-ellipsis"
    >
      <svg
        aria-hidden="true"
        class=""
        data-icon="ellipsis"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M176 511a56 56 0 1 0 112 0 56 56 0 1 0-112 0zm280 0a56 56 0 1 0 112 0 56 56 0 1 0-112 0zm280 0a56 56 0 1 0 112 0 56 56 0 1 0-112 0z"
        />
      </svg>
    </i>
  </button>
</div>
`;
